src/ngraph/op/constant.hpp

```diff
@@ -240,8 +255,58 @@ namespace ngraph
             }
 
             element::Type m_element_type;
-            Shape m_shape;
-            void* m_data;
+            Shape m_shape{};
+            void* m_data{nullptr};
+        };
+
+        class ScalarConstantLikeBase : public Constant
+        {
+        public:
+            std::shared_ptr<op::Constant> as_constant() const;
+
+        protected:
+            ScalarConstantLikeBase(const std::string& name, const NodeVector& args)
+                : Constant(name, args)
+            {
+            }
+        };
+
+        /// \brief A scalar constant whose element type is the same as like.
+        template <typename T>
+        class ScalarConstantLike : public ScalarConstantLikeBase
```

Comment:
I'm kind of leery of a templated op class here. We've managed to avoid it for `Constant`.

Reply:
The problem is that we don't know the element type until later, so we need a way to hold onto the original value until we know the element type.